tensorflow/core/common_runtime/process_util.cc

```diff
@@ -49,8 +51,22 @@ thread::ThreadPool* ComputePool(const SessionOptions& options) {
 int32 NumInterOpThreadsFromSessionOptions(const SessionOptions& options) {
   const int32 t = options.config.inter_op_parallelism_threads();
   if (t != 0) return t;
+#ifdef INTEL_MKL
+  // MKL library executes ops in parallel using OMP threads
+  // Set inter_op conservatively to avoid thread oversubscription that could 
+  // lead to severe perf degradations and OMP resource exhaustion
+  const int mkl_intra_op = omp_get_max_threads();
+  CHECK_GE(mkl_intra_op, 1);
+  const int32 mkl_inter_op = std::ceil(
+    static_cast<float>(port::NumSchedulableCPUs()) / mkl_intra_op);
```

Comment:
There is no reason to go to float just to round up.

(port::NumSchedulableCPUs() + mkl_intra_op - 1) / mkl_intra_op

does what you want. This is also defined as inline function div_up or similar, in several places.

Reply:
will do.